nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/action/computation/SortBlob.java

```diff
@@ -50,6 +55,32 @@ public void processRecord(ComputationContext context, String inputStreamName, Re
         String commandId = in.getCommandId();
         Blob blob = getBlob(in.getDataAsString());
         blob = sort(blob, commandId);
+
+        File blobFile = blob.getFile();
+        try {
+            BufferedReader br = new BufferedReader(new FileReader(blobFile));
+            // Add header
+            String header = in.getHeaderAsString();
+            StringBuilder result = StringUtils.isNotEmpty(header) ? new StringBuilder(String.format("%s%n", header))
+                    : new StringBuilder();
+            String line;
+            while ((line = br.readLine()) != null) {
+                result.append(String.format("%s%n", line));
```

Comment:
you should handle the file as binary, just create a new file, append the header as binary, then open an inputstream on the blob and append the content to the file and the footer.

Reply:
Agreed, it'll be much faster than doing line-level reading.